cognite/client/_api_client.py

```diff
@@ -418,79 +427,147 @@ def _list_generator(
             if method == "GET":
-                params = filter.copy()
-                params["limit"] = current_limit
-                params["cursor"] = next_cursor
-                if sort is not None:
-                    params["sort"] = sort
-                params.update(other_params or {})
-                res = self._get(url_path=url_path or resource_path, params=params, headers=headers)
-
-            elif method == "POST":
-                body: dict[str, Any] = {"limit": current_limit, "cursor": next_cursor, **(other_params or {})}
-                if filter:
-                    body["filter"] = filter
-                if advanced_filter:
-                    body["advancedFilter"] = (
-                        advanced_filter.dump(camel_case_property=True)
-                        if isinstance(advanced_filter, Filter)
-                        else advanced_filter
-                    )
-                if sort is not None:
-                    body["sort"] = sort
-                res = self._post(
-                    url_path=url_path or resource_path + "/list",
-                    json=body,
-                    headers=headers,
-                    api_subversion=api_subversion,
-                )
+                res = self._get(url_path=url_path, params=params, headers=headers)
             else:
-                raise ValueError(f"_list_generator parameter `method` must be GET or POST, not {method}")
-            last_received_items = res.json()["items"]
-            total_items_retrieved += len(last_received_items)
+                res = self._post(url_path=url_path, json=params, headers=headers, api_subversion=api_subversion)
 
-            if not chunk_size:
-                for item in last_received_items:
-                    yield resource_cls._load(item, cognite_client=self._cognite_client)
-            else:
-                unprocessed_items.extend(last_received_items)
-                if len(unprocessed_items) >= chunk_size:
-                    chunks = split_into_chunks(unprocessed_items, chunk_size)
-                    if chunks and len(chunks[-1]) < chunk_size:
-                        unprocessed_items = chunks.pop(-1)
-                    else:
-                        unprocessed_items = []
-                    for chunk in chunks:
-                        yield list_cls._load(chunk, cognite_client=self._cognite_client)
-
-            next_cursor = res.json().get("nextCursor")
-            if total_items_retrieved == limit or next_cursor is None:
-                if chunk_size and unprocessed_items:
+            response = res.json()
+            yield from self._process_into_chunks(response, chunk_size, resource_cls, list_cls, unprocessed_items)
+
+            next_cursor = response.get("nextCursor")
+            total_retrieved += len(response["items"])
+            if total_retrieved == limit or next_cursor is None:
+                if unprocessed_items:  # can only happen when using chunk_size
```

Comment:
Maybe an extra sentence why this can only happen when using chunk_size. 

Is it become the `_process_into_chunck` will take care of it?

Reply:
I'll change to:
# only happens when not yielding one-by-one

Reply:
That helps